crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,409 @@
+    match item {
+        AnyCssDeclarationOrRule::CssBogus(_) => SortInfo::unknown(),
+        AnyCssDeclarationOrRule::CssMetavariable(_) => SortInfo::unknown(),
+        AnyCssDeclarationOrRule::AnyCssRule(rule) => match rule {
+            AnyCssRule::CssAtRule(_) => SortInfo::from_kind(SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE),
+            AnyCssRule::CssBogusRule(_) => SortInfo::unknown(),
+            AnyCssRule::CssNestedQualifiedRule(_) => {
+                SortInfo::from_kind(SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE)
+            }
+            AnyCssRule::CssQualifiedRule(_) => SortInfo::unknown(),
+        },
+        AnyCssDeclarationOrRule::CssDeclarationWithSemicolon(decl_with_semicolon) => {
+            let prop = decl_with_semicolon
+                .declaration()
+                .ok()
+                .and_then(|decl| decl.property().ok());
+
+            if let Some(prop) = prop {
+                match prop {
+                    AnyCssProperty::CssComposesProperty(_) => {
+                        SortInfo::from_kind(SORT_INFO_KIND_COMPOSES_PROPERTY)
+                    }
+                    AnyCssProperty::CssGenericProperty(prop) => match prop.name().ok() {
+                        Some(name) => match name {
+                            AnyCssDeclarationName::CssDashedIdentifier(_) => {
+                                SortInfo::from_kind(SORT_INFO_KIND_CUSTOM_PROPERTY)
+                            }
+                            AnyCssDeclarationName::CssIdentifier(ident) => {
+                                if let Some(sanitized_prop) = css_identifier_to_prop_name(&ident) {
+                                    let (vendor_prefix, plain_prop) = sanitized_prop;
+
+                                    let vendor_prefix_idx: u32 = VENDOR_PREFIXES
+                                        .iter()
+                                        .position(|vp| vp == &vendor_prefix.as_str())
+                                        .map_or(u32::MAX, |pos| pos as u32);
+
+                                    if let Some(idx) = PROPERTY_ORDER_MAP.get(&plain_prop) {
+                                        SortInfo::from_declaration(*idx, vendor_prefix_idx)
+                                    } else {
+                                        SortInfo::from_kind(SORT_INFO_KIND_UNKNOWN_DECLARATION)
+                                    }
+                                } else {
+                                    SortInfo::unknown()
+                                }
+                            }
+                        },
+                        None => SortInfo::unknown(),
+                    },
+                    _ => SortInfo::unknown(),
+                }
+            } else {
+                SortInfo::unknown()
+            }
+        }
+    }
+}
+
+fn contains_shorthand_after_longhand(items: &[NodeWithPosition]) -> bool {
+    let mut seen_shorthand_properties = HashSet::<String>::with_capacity(items.len());
```

Comment:
Using `with_capacity` isn't very efficient in this case, because inside the loop there are cases of early return e.g. `return true`. This means that you create a list where some slots will be always empty, and "wasted".